src/core/utils/props.ts

```diff
@@ -26,3 +28,40 @@ export type CommonProps = {
    */
   id?: string;
 } & StylingProps;
+
+/**
+ * Merges provided Props with the props of T.
+ *
+ * T can be any native HTML element or a custom component.
+ */
+export type PolymorphicComponentProps<
```

Comment:
Don't we need `Omit` here? Saw in a video:
![image](https://user-images.githubusercontent.com/36186912/148531098-17a5570b-7817-44c5-8baa-add1935e3b7e.png)
https://www.youtube.com/watch?v=uZ8GZm5KEXY

Reply:
The merge utility takes care of that 
![image](https://user-images.githubusercontent.com/9084735/148563939-52451d1d-0e45-4b26-b235-5f184e3b5a30.png)